### ISSUE TITLE
Add "format" to options (cannot paste tab-delimited data into excel)

### ISSUE THREAD

**Issue by gravitypersists** [sudodoki/copy-to-clipboard#47]
I came across this issue when trying to copy some tab-delimeted data to be paste-able into excel. I needed to paste my contents as `text/plain` because I guess that wasn't the default the browser does.

According to [the w3 spec](https://www.w3.org/TR/clipboard-apis/#the-copy-action), you can do this like so:

```
document.addEventListener('copy', function(e){
    e.clipboardData.setData('text/plain', 'Hello, world!');
    e.clipboardData.setData('text/html', '<b>Hello, world!</b>');
    e.preventDefault(); // We want our data, not data from any selection, to be written to the clipboard
});
```

`clipboardData` is a [DataTransfer](https://html.spec.whatwg.org/multipage/interaction.html#dom-datatransfer-setdata) object so it's `setData` *should* be well-supported, but I'm not sure about the document's copy event being very well supported.

I'd be happy to implement these changes, but I thought I'd raise an issue first to make sure this is something you'd be okay with adding to the library.

Ideally, the fix would be along the lines of: `copy('blah blah', { debug: true, format: 'text/plain' })`

**Comment by sudodoki** [comment]
Let me think about it. Thanks for bringing this up! 🙇

**Comment by dpwrussell** [comment]
@gravitypersists Do you have an implementation of this? I ran into this exact same problem for the exact same use-case today.

**Comment by gravitypersists** [comment]
Yeah, the solution was to add the document listener for 'copy' as I detailed in the original text of the issue. I think it circumvents a bit of what this lib does, so you might lose some checks and, like I said, not sure how well 'copy' is implemented in all browsers (I have the luxury of only needing to worry about modern desktop browsers).

**Comment by dpwrussell** [comment]
@gravitypersists Thanks for your response, I was able to get that working.

@sudodoki I think this could be integrated into your library by adding an option which would (where possible) do the: Add listener -> Copy -> Remove listener. Perhaps simply providing a function as the listener to `copy`as an option would trigger this behaviour and if possible this would be called.

I guess the negative is that the other fallback copy methods would not be exactly the same as this, but I'm not sure there is anything you can do about that so as long as it is _caveat emptor_, then I'd say this would be a worthwhile addition.

**Comment by ghostCoder** [comment]
Any update on this?  This has become a blocker for me :(

**Comment by sudodoki** [comment]
so sorry to hear this. I'm not actively working with this library and cannot pay too much time for this. Maybe, if @gravitypersists stands up I could just kindly ask him to implement this and make sure it all works out in the end 😅

**Comment by ghostCoder** [comment]
yeah @gravitypersists 's solution worked for me. thanks!

**Comment by ecwyne** [comment]
Has anyone tried doing this to copy-paste html into an iOS email. I can use `copy` no problem, but when I tinker with `clipboardData.setData()` at all (even setting it to `text/plain`) I'm not able to paste into the email app.

**Comment by MarceloBarrera** [comment]
@gravitypersists 's solution worked for me but does not work on IE11 because it does not support 'text/plain' but 'Text' instead. The issue is not only with tabs but line breaks as well when pasting to an excel sheet.

**Comment by sudodoki** [comment]
I guess this landed after #79 although not sure about IE11 specific stuff correctness
